addons/web_editor/static/src/js/editor/snippets.options.js

```diff
@@ -5922,6 +5922,9 @@ const ImageHandlerOption = SnippetOptionWidget.extend({
             // Convert to recommended format and width.
             img.dataset.mimetype = 'image/webp';
             img.dataset.resizeWidth = this.optimizedWidth;
+        } else if (img.dataset.shape) {
+            img.dataset.originalMimetype = "image/webp";
+            img.dataset.resizeWidth = this.optimizedWidth;
```

Comment:
You should still exclude the gif here, because of the following situation:
- Drop Text-Image
- Add a shape on the image
- Replace the image by a gif 

=> The gif is converted to webp (which should not be the case I think) and is therefore not animated anymore

Reply:
Ho, indeed, your are right, sorry about, I also should have think about it myself :cry: 
@qsm-odoo, for a case like this, is it better to solve the problem in master and then the fix will only be for the previous version or is it better to let it like the previous version and the fix will target both previous and current versions?

Reply:
@loco-odoo It does not matter, as you wish :)

Personally, I would fix the bug in this PR, then make a fix 16.4 only. That way, less work for our runbot/mergebot and a cleaner branch history in master.